lib/nats.js

```diff
@@ -891,7 +890,6 @@ Client.prototype.processInbound = function() {
                             // if we have a stream, this is from an old connection, reap it
                             if (client.stream) {
                                 client.stream.removeAllListeners();
-                                client.stream.end();
```

Comment:
You had introduced the call to `.end()` in some commits (https://github.com/nats-io/node-nats/commit/be8fa847de9691297f679d7ab7044411795fc543). Could you try recall what problem you tried to fix and that this is not going to re-introduce the issue? If there was a test added at the time and it still pass, I guess no.

Reply:
@kozlovic see below

Reply:
Sorry, I missed the fact that we pass the socket as part of the options to `tls.Connect()`.